homeassistant/components/neato/camera.py

```diff
@@ -20,11 +30,15 @@
 ATTR_GENERATED_AT = "generated_at"
 
 
-async def async_setup_entry(hass, entry, async_add_entities):
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
     """Set up Neato camera with config entry."""
     dev = []
-    neato = hass.data.get(NEATO_LOGIN)
-    mapdata = hass.data.get(NEATO_MAP_DATA)
+    neato: NeatoHub | None = hass.data.get(NEATO_LOGIN)
```

Comment:
You could use `Optional[NeatoHub]` instead of `NeatoHub | None`. I think that's a bit cleaner

Reply:
If I esclude casts, I see only 2 usages of Optional in the whole components folder

Reply:
So what? It doesn't matter performance wise and optional is easier to read and understand. What's the point of arguing here?

Reply:
I don't want to argue with you, I'm just saying that the general usage in core is based on the syntax used in this PR.

Reply:
Is there a reason not to use optional? Just because it's not used by others is not enough of a reason. That's my personal view, of course. Do you think ` | None` is easier to read?

Reply:
Yes, I prefer `| None`, but I ask a second opinion so we can easily sort it.

Reply:
We've replaced the use of `Optional` with `| None` in the whole application.